waku/v2/protocol/filter/client.go

```diff
@@ -236,11 +203,9 @@ func (wf *WakuFilterLightNode) request(ctx context.Context, params *FilterSubscr
 
 // Subscribe setups a subscription to receive messages that match a specific content filter
 func (wf *WakuFilterLightNode) Subscribe(ctx context.Context, contentFilter ContentFilter, opts ...FilterSubscribeOption) (*SubscriptionDetails, error) {
-	wf.RLock()
-	defer wf.RUnlock()
-
-	if !wf.started {
-		return nil, errNotStarted
+	defer wf.RLockAndUnlock()()
```

Comment:
This is probably subjective, but I find `wf.RLockAndUnlock()()` less readable than using `wf.RLock()` followed by `defer wf.RUnlock()`, which is a much more common pattern

Reply:
The function is actually not unlocking the mutex rather returning another function which can be used to unlock.
I agree with @richard-ramos , Let us just use Rlock and RUnlock. Would make reading code easier.